### DISCUSSION TITLE
Question about SRP and IPPS.

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#8157] [Q&A] [answered]
I would like to ask you some questions regarding SRP and IPPS.

In this tutorial (https://openthread.io/codelabs/openthread-border-router), Section 5 explains how to advertize the presence of the Thread device through SRP. The following commands are used:

```
srp client autostart enable
srp client host name ot-host
srp client host address fd76:a5d1:fcb0:1707:d3dc:26d3:f70b:b927
srp client service add ot-service _ipps._tcp 12345
```

which corresponds to the following names:
```
SRP_CLIENT_HOST_NAME       "ot-host"
SRP_SERVICE_INSTANCE_NAME  "ot-service"
SRP_SERVICE_NAME           "_ipps._tcp"
```

I'm using Service Browser on my Android phone and I see several kind of services in my home. For example I have:
- "_hap._tcp" (= Host Access Protocol)
- "_canon_bjnp1._tcp"  (another service for my printer)
- ...etc

In our Thread app, I think that we can use whatever names we want for the host name ("ot-host") and the service instance name ("ot-service"). But what about the service name? Do we have to use "_ipps._tcp" or is it just an example?
I don't understand what makes this service an Internet Printing Protocol service. 

In my application, I use names like this (where XXXX is a unique Id):
```
SRP_CLIENT_HOST_NAME       "ot-host-XXXXXXX"
SRP_SERVICE_INSTANCE_NAME  "ot-service-XXXXXXX"
SRP_SERVICE_NAME           "_ipps._tcp"
```
Shall I also ensure that the service name ("_ipps._tcp") is unique? 
And can I use another name like "_my_app_name._tcp"? 

Thank you

**Comment by jwhui** [comment, marked as answer]
> But what about the service name? Do we have to use "_ipps._tcp" or is it just an example?

It is just an example. You may use what ever names best fits your needs.

> I don't understand what makes this service an Internet Printing Protocol service.

It is not at all related to Internet Printing Protocol - it is just an example.

> Shall I also ensure that the service name ("_ipps._tcp") is unique?

The service name need not be unique, but the service instance name does need to be unique.

> And can I use another name like "_my_app_name._tcp"?

Yes.

Feel free to experiment and try different values!

**Reply by OlivierGre** [reply]
Perfect, thank you Jonathan
